spec/core_ext/string/decimal_suffix_spec.rb

```diff
@@ -1,21 +1,51 @@
 describe String do
-  it '#decimal_si_to_f' do
-    expect("1f".decimal_si_to_f).to eq(0.000_000_000_000_001)
-    expect("1p".decimal_si_to_f).to eq(0.000_000_000_001)
-    expect("1n".decimal_si_to_f).to eq(0.000_000_001)
-    expect("1μ".decimal_si_to_f).to eq(0.000_001)
-    expect("1m".decimal_si_to_f).to eq(0.001)
-    expect("1c".decimal_si_to_f).to eq(0.01)
-    expect("1d".decimal_si_to_f).to eq(0.1)
-    expect("1".decimal_si_to_f).to  eq(1.0)
-    expect("1k".decimal_si_to_f).to eq(1_000.0)
-    expect("1M".decimal_si_to_f).to eq(1_000_000.0)
-    expect("1G".decimal_si_to_f).to eq(1_000_000_000.0)
-    expect("1T".decimal_si_to_f).to eq(1_000_000_000_000.0)
-    expect("1P".decimal_si_to_f).to eq(1_000_000_000_000_000.0)
-    expect("1E".decimal_si_to_f).to eq(1_000_000_000_000_000_000.0)
+  it '#decimal_si_to_f', :aggregate_failures => true do
+    # All these are not exactly equal if computed via integer * factor,
+    # e.g. 87 * 0.001 != 0.087.
+    expect("93f".decimal_si_to_f).to eq(0.000_000_000_000_093)
+    expect("92p".decimal_si_to_f).to eq(0.000_000_000_092)
+    expect("93n".decimal_si_to_f).to eq(0.000_000_093)
+    expect("91μ".decimal_si_to_f).to eq(0.000_091)
```

Comment:
Is there a reason you're not using the same number for all of the tests?

Reply:
I wanted test cases that all failed with previous algorithm.  This happens for different values semi-randomly, depending on how powers of 2 and 10 align...
```
[7] pry(main)> 93 * 1e-12
=> 9.3e-11
[8] pry(main)> 92 * 1e-12
=> 9.199999999999999e-11
[9] pry(main)> 92 * 1e-9
=> 9.2e-08
[10] pry(main)> 93 * 1e-9
=> 9.300000000000001e-08
```

Reply:
Should we change these tests back to the original format and add a new test verifying several values that used to fail for rounding errors or other issues?  I see them as two separate issues and wouldn't want to lose this test because someone modified these expectations to consistently use the same number.

Reply:
I don't see a point having both old and new test, I view 1 as just a particularly easy special case (even changing 1 to 3 makes a couple fail).
I'm not sure if this is what you had in mind, but rewrote to test all input forms over all digit pairs. Total 192/700 failures on the fractional suffixes, all passing after 2nd commit.
PTAL.